internal/service/resources/cmd/serve.go

```diff
@@ -1,25 +1,98 @@
 package cmd
 
 import (
+	"fmt"
 	"log/slog"
 	"os"
 
-	"github.com/openshift-kni/oran-o2ims/internal/service/resources"
 	"github.com/spf13/cobra"
+
+	"github.com/openshift-kni/oran-o2ims/internal/cmd/server"
+	"github.com/openshift-kni/oran-o2ims/internal/controllers/utils"
+	"github.com/openshift-kni/oran-o2ims/internal/service/resources"
+	"github.com/openshift-kni/oran-o2ims/internal/service/resources/api"
 )
 
+// config defines the configuration attributes for the resource server
+var config api.ResourceServerConfig
+
 // resourceServer represents start command for the resource server
 var resourceServer = &cobra.Command{
 	Use:   "serve",
 	Short: "Start resource server",
 	Run: func(cmd *cobra.Command, args []string) {
-		if err := resources.Serve(); err != nil {
+		if err := resources.Serve(&config); err != nil {
 			slog.Error("failed to start resource server", "err", err)
 			os.Exit(1)
 		}
 	},
 }
 
+// setServerFlags creates the flag instances for the server
+func setServerFlags(cmd *cobra.Command) {
+	flags := cmd.Flags()
+	flags.StringVar(
+		&config.Listener.Address,
+		server.APIListenerAddressFlagName,
+		"127.0.0.1",
+		"API listener address.",
+	)
+	flags.StringVar(
+		&config.Listener.CertFile,
+		server.APIListenerTLSCertFlagName,
+		"/secrets/tls/tls.crt",
+		"API TLS certificate in PEM format.",
+	)
+	flags.StringVar(
+		&config.Listener.KeyFile,
+		server.APIListenerTLSKeyFlagName,
+		"/secrets/tls/tls.key",
+		"API TLS key in PEM format.",
+	)
+	flags.StringVar(
+		&config.CloudID,
+		server.CloudIDFlagName,
+		"",
+		"The local O-Cloud identifier.",
+	)
+	flags.StringVar(
+		&config.BackendURL,
+		server.BackendURLFlagName,
+		"",
+		"URL of the backend search-api component.",
+	)
+	flags.StringArrayVar(
+		&config.Extensions,
+		server.ExtensionsFlagName,
+		[]string{},
+		"Extensions to add to resources and resource pools.",
+	)
+	flags.StringVar(
+		&config.ExternalAddress,
+		server.ExternalAddressFlagName,
+		"string",
+		"URL used to access the O-Cloud Manager from outside of the cluster.",
+	)
+	flags.StringVar(
+		&config.GlobalCloudID,
+		server.GlobalCloudIDFlagName,
+		utils.DefaultOCloudID,
+		"The global O-Cloud identifier.",
+	)
+
+	// The O-Cloud ID and External address arguments are mandatory while all other arguments are optional.  The Global
+	// O-Cloud ID is special in that it is not strictly mandatory to start the server but it is mandatory to enable
+	// some API endpoints (e.g., subscriptions).
```

Comment:
why is it needed to enable some API endpoints?

Reply:
Because the global cloud id is a required piece of data to send notification events to subscribers and so I don't believe we should allow a new subscription to be created until the end user has provided us with our global cloud id.   Our servers need to be up and running before the user will have provided this so we can't make it mandatory.  The best we can do is block some API endpoints if it is not provided -- with a 400 bad request with a suitable message. 

It appears in both the alarm event and the cloud available notification events.   I think I originally confused the cloud available event with the regular inventory change notifications so it isn't in fact needed on the resource server API endpoints, but it is necessary on the alarm server subscription create API.  ...but I'm sort of the view that if we block the alarm subscriptions we should block the inventory subscriptions as well to create consistent behaviour.

Reply:
Why do resource and alarms servers need to up and running before the user has provided the global cloud id? Even after they do, we will need to restart the servers, or not?

How is this (global cloud id) provided? (haven't read the code for that part yet) Shouldn't be mandatory by the operator?